### PR TITLE
Update to single package repo and add GitHub Action

### PR DESCRIPTION
**Pull request by rajsite**
- Moves the `@ni/eslint-config` package to the root of the repository
- Adds pull request and branch build GitHub Action workflows
- Adds release GitHub Action workflow for tagged commits

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+on:
+  pull_request_target:
```
Comment (rajsite) [inline .github/workflows/build.yml:2]: Enables safe builds for fork-pr workflows for contributors from outside the repository

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+on:
+  push:
+    tags:
+    - 'v*'
```
Comment (rajsite) [inline .github/workflows/release.yml:4]: Release on tag commits created by the `npm version` command

File: README.md [hunk 3]
```diff
@@ -1,20 +1,20 @@
 # NI JavaScript and TypeScript Style Guide
 
-![Logo](resources/logo.svg)
+![Logo](https://raw.githubusercontent.com/ni/javascript-styleguide/HEAD/resources/logo.svg)
```
Comment (rajsite) [inline README.md:3]: Update logo usage in README to point to github served content to make visilbe on npmjs.com README

File: package.json [hunk 4]
```diff
@@ -1,24 +1,42 @@
...
   "bugs": {
     "url": "https://github.com/ni/javascript-styleguide/issues"
   },
-  "homepage": "https://github.com/ni/javascript-styleguide#readme"
+  "homepage": "https://github.com/ni/javascript-styleguide#readme",
+  "publishConfig": {
+    "access": "public"
+  },
+  "files": [
+    ".eslintrc.js"
```
Comment (rajsite) [inline package.json:29]: Add explicit file include list so other files added to repo are ignored (License, README, pacakge.json automatically pulled in by npm)

File: .github/workflows/build.yml (hunk 1 above)
Comment (TrevorKarjanis) [reply .github/workflows/build.yml:2]: This is effectively saying when a pull request is opened or pushed to run the build job in a way that is compatible with forks?

File: README.md (hunk 3 above)
Comment (TrevorKarjanis) [reply README.md:3]: Understood!

File: README.md [hunk 5]
```diff
@@ -1,20 +1,20 @@
...
 
-Use [npm view](https://docs.npmjs.com/cli/view.html) to get the correct versions of each package.
+Use [npm view](https://docs.npmjs.com/cli/view.html) to list the correct versions of each package to install yourself.
 
 ```bash
 npm view @ni/eslint-config peerDependencies
 ```
 
-Alternatively, use a shortcut with [install-peerdeps](https://www.npmjs.com/package/install-peerdeps).
+Alternatively, use [install-peerdeps](https://www.npmjs.com/package/install-peerdeps) as a shortcut to install the pakages for you.
```
Comment (TrevorKarjanis) [inline README.md:17]: Packages is misspelled.

File: package.json (hunk 4 above)
Comment (TrevorKarjanis) [reply package.json:29]: Does npmjs.com use the GitHub README or the package's? The [documentation](https://docs.npmjs.com/about-package-readme-files) just says, "we recommend including a README file in your package directory."

**Review by TrevorKarjanis — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (rajsite) [reply README.md]: fixed!

File: package.json (hunk 4 above)
Comment (rajsite) [reply package.json:29]: Yep! The linked page states "The README file will be shown on the package page."

File: .github/workflows/build.yml (hunk 1 above)
Comment (rajsite) [reply .github/workflows/build.yml:2]: Yep! Specifically the [pull_request_target](https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull_request_target) configuration uses the target branch GitHub Actions configuration. That way a drive-by user cannot create a PR that modifies the workflow actions to expose secrets in the build.

**Review by TrevorKarjanis — APPROVED** [review]

**Pull request merged**